lib/Transforms/LazyRelin/LazyRelin.td

```diff
@@ -0,0 +1,26 @@
+#ifndef LIB_TRANSFORMS_LAZY_RELIN_TD_
+#define LIB_TRANSFORMS_LAZY_RELIN_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def LazyRelin : Pass <"lazy-relin"> {
+    let summary = "Apply lazy relinearization";
+    let description = [{
+        This pass defers relinearization ops as late as possible in the IR.
+        This is more efficient in cases where multiplication operations are followed by
+        additions, such as in a dot product. Because relinearization also adds error,
+        deferring it can reduce the need for bootstrapping.
+
+        In much of the literature, lazy relinearization is applied manually. See for example
+        Blatt-Gusev-Polyakov-Rohloff-Vaikuntanathan 2019 https://eprint.iacr.org/2019/223
+        and Lee-Lee-Kim-Kim-No-Kang 2020 https://eprint.iacr.org/2020/1549.
```

Comment:
I think it'd be good to also mention EVA/EVA Improved and other work that looks at automatically inserting relin.

Reply:
Done in ef66e8eb

This also makes me think we should rename the pass to not be called "lazy", since "lazy" implies a specific kind of heuristic.